### ISSUE TITLE
Determine if utterance audio contained only silence or babble

### ISSUE THREAD

**Issue by aargreen** [Azure-Samples/cognitive-services-speech-sdk#1066]
**Is your feature request related to a problem? Please describe.**
I would like to track the reasons why the microsoft speech sdk gives no results.  I have been able to differentiate some categories of errors (like no Language Identified), but I would like to separate out utterances where the audio sent was all silence or all babble. I have tried using the NoMatchDetails class but that seems to give "NoMatch" rather than "SilenceTimeout" even for the case of pure silence.  My guess is my users are perhaps terminating the audio stream prior to the Microsoft SDK Silence Timeout time being reached, but this is just speculation as to why it does return Silence TImeout for this care.

Is there a API/ way to determine if the Recognizer only received silence or babble at the time of a NoMatch event?

**Comment by amitkumarshukla** [comment]
@aargreen Thanks a lot for reporting this issue. I can see the similar behavior with the speech sdk. We will investigate it further and will get back to you as soon as possible.

**Comment by aargreen** [comment]
Hi @amitkumarshukla ,  Sorry should have specified, this  is in java.  
I did attempt to use NoMacthDetails in:
recognizer.recognized.addEventListener((s, e) {}

Within the event lister this is this line of code: if (e.getResult().getReason() == ResultReason.NoMatch)

I got the NoMatchDetails just after that, but even for pure silence the NoMatchDetails reported "NoMatch", not "SilenceTimeout" or "BabbleTimeout".  

My thought was perhaps the reason for this is if the audio stream stops before a certain period of time it doesn't trigger these returns.  If that's the case, was essentially wondering if there is a way to get the information that the audio stream to date was either silence or babble up to the point of the stream ending (to differentiate that from someone saying "hello", for example, and then the stream ending)

**Comment by amitkumarshukla** [comment]
@aargreen In Java you need to do the following 
```java
                if(e.getResult().getReason() == ResultReason.NoMatch){
                    System.out.println("No match reason is : " + NoMatchDetails.fromResult(e.getResult()));
                }
```

Doing that I can see the initial silence timeout. 
By the way by default the initial silence timeout is 30 seconds. So if your stream is less than 30 seconds you will not see the initial silence timeout.

**Comment by aargreen** [comment]
@amitkumarshukla The question is, let's say I have a stream which is 3 seconds long and entirely silence.  Is it possible to get an indication from the SDK that the stream contained only silence?  

In general I am attempting to separate out failures to return a response for a "reasonable input" vs. an "unreasonable" input.    So for example I would like to separate the case where the user says "Hello" but perhaps is difficult to hear (or its perhaps difficult to guess at their language) vs a case where the user submitted only silence.

**Comment by amitkumarshukla** [comment]
@aargreen please follow the thread https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/502 and let me know if that helps with your question ?

**Comment by aargreen** [comment]
@amitkumarshukla I don't think that thread answers my question if I am understanding it correctly.    The main thing I take away from that thread is the audio segmentation is fairly complicated.   

In my question I don't really want to affect the audio segmentation.. just want to know if a segment is just silence. (or perhaps just babble).  

There may not be a way to receive this information.. but just wanted to check on it.  If we are confident there is not a way to receive the information, I'm ok with closing the issue (or leave it open as a feature request).
